File: project 4/c++/task2.cpp
 #include <iostream>

using namespace std;

int add(int a){
    // add 6 to input
    return a+6;
}

int main()
{
   cout << "Demo" << endl; 
   
   // declare function variable
   auto function=add;
   
   // demonstrate function used as data type
   cout << "Output of func with 16 input :"<<function(16) << endl; 
   // print out type id
   cout << " function data type id :" <<typeid(function).name()<< endl; 
        
   return 0;
}

File: project 4/Ruby/task2.rb
# Ruby demo
puts "Ruby function as data type \n\n";

a=10
b=13
c=-5

# my function
def add(x)
    return x+7;
end

def multiply(y)
    return y*4;
end


# assigning a function onto a variable, that gives us an error
#func=add
# we can however use the method 
func_add=method(:add);
func_mult=method(:multiply);

# calling this variable declared functions

puts "Add 7 to #{a} gives #{func_add.call(a)}\n"
puts "Multiplying by  #{a} by 4  gives #{func_mult.call(a)}\n"


# assinging function output to a variable
output=add(10)
puts "Assinging function out put to a variable, adding 7 to our input 10, we get #{output}\n"

# passing function as input into another function possible
d=add(multiply(6))
puts "Making function input to another function, we get #{d}";


















